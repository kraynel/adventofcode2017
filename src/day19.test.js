const { getLetters } = require("./day19");
const fs = require("fs");

describe("getLetters", () => {
  it("should getLetters", () => {
    expect(
      getLetters(`     |          
     |  +--+    
     A  |  C    
 F---|----E|--+ 
     |  |  |  D 
     +B-+  +--+`).letters
    ).toBe("ABCDEF");
  });

  it("should get the value", done => {
    fs.readFile("inputs/day19.txt", "utf8", (err, data) => {
      expect(getLetters(data).letters).toMatchSnapshot();
      done();
    });
  });
});

describe("getSteps", () => {
  it("should getLetters", () => {
    expect(
      getLetters(`     |          
     |  +--+    
     A  |  C    
 F---|----E|--+ 
     |  |  |  D 
     +B-+  +--+`).steps
    ).toBe(38);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day19.txt", "utf8", (err, data) => {
      expect(getLetters(data).steps).toMatchSnapshot();
      done();
    });
  });
});
