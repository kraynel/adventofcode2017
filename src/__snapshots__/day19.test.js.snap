// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLetters should get the value 1`] = `"VTWBPYAQFU"`;

exports[`getSteps should get the value 1`] = `17358`;
